geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/partitions/DateTimeScheme.scala

```diff
@@ -44,20 +48,54 @@ case class DateTimeScheme(format: String, stepUnit: ChronoUnit, step: Int, dtg:
       dt => dt.`with`(adjuster).truncatedTo(DAYS)
 
     case _ =>
-      dt => ZonedDateTime.parse(fmt.format(dt), fmt) // fall back to format and re-parse
+      dt => DateParsing.parse(fmt.format(dt), fmt) // fall back to format and re-parse
   }
 
   // TODO This may not be the best way to calculate max depth...
   // especially if we are going to use other separators
-  override val depth: Int = format.count(_ == '/')
+  override val depth: Int = format.count(_ == '/') + 1
```

Comment:
*blinks* where did the +1 come from?

Reply:
this was incorrect, but i think wasn't ever really used. if you think about say a single yearly partition, your pattern is 'yyyy' - depth in that case would be 1, not 0. maybe there are some corner cases here in how you define your pattern with leading/trailing slashes? but at least for the built-in ones, this works correctly.